### PR TITLE
[1.16] Optionally add milk as a fluid to the vanilla bucket

### LINKED ISSUE #6769: [1.15.2][1.16] Milk Bucket Fluid Wrapper forces the vanilla milk bucket to return no fluid
**Minecraft Version:** 1.15.2, 1.16.1

**Forge Version:** 31.2.9

**Description of issue:**

The Forge fluid bucket wrapper is currently hardcoded to only support `BucketItem`, however it is returned for the vanilla milk bucket. As a result, that bucket always [returns empty](https://github.com/MinecraftForge/MinecraftForge/blob/1.15.x/src/main/java/net/minecraftforge/fluids/capability/wrappers/FluidBucketWrapper.java#L71-L82). Normally this would not be a problem as we can just attach a new capability using the event, but the native capability provider [run before event added ones](https://github.com/MinecraftForge/MinecraftForge/blob/1.15.x/src/main/java/net/minecraftforge/common/capabilities/CapabilityDispatcher.java#L70-L78), causing the Forge one to win out. This means there is no way to add a milk fluid as a return for the bucket save a registry override on the milk bucket item.

[Other places](https://github.com/MinecraftForge/MinecraftForge/blob/1.15.x/src/main/java/net/minecraftforge/fluids/capability/wrappers/FluidBucketWrapper.java#L61-L68) in the code hardcode the fluid [to fill the milk bucket](https://github.com/MinecraftForge/MinecraftForge/blob/9114bec81d2bb705f8dbaa0ab70e4673fbfbb288/src/main/java/net/minecraftforge/fluids/FluidUtil.java#L644-L647) to `minecraft:milk`, which is equally bad as mods should not be registering fluids to the Minecraft namespace

**Solution:**

The logic in its current state is pretty useless, as there is no way for mods to extend it and the only remaining logic is redundant to the [default attribute getter](https://github.com/MinecraftForge/MinecraftForge/blob/9114bec81d2bb705f8dbaa0ab70e4673fbfbb288/src/main/java/net/minecraftforge/fluids/FluidUtil.java#L650). The logic could be fixed to actually be useful, such as using the `forge:milk` tag to find the proper fluid to return, but I think such a solution is overkill. The easiest solution I see is just removing all Forge patches related to the milk bucket fluid container, as its pretty easy for the mod adding milk to add their own capability to milk buckets.

I can make a pull request if desired, once I know the preferred solution. As I said, I am in favor of reducing patches and logic by just leaving milk buckets as not a fluid container unless a mod adds a capability.

### PR DESCRIPTION
**Pull request by KnightMiner**
This pull request supersedes #6840, in order to fix #6769. After discussion on Discord, it was determined the best solution for mods to add milk as a fluid was for Forge to register the fluid, while mods just set a Forge flag to enable it, similar to the universal bucket back in 1.12

To enable Forge adding milk, you simply need to call `ForgeMod.enableMilkFluid()` during mod construction. We can move this function to a different location if there is a better one, but this location kept the logic behind the flag clean.

A few details on implementation:
* It was suggested on Discord that we use `minecraft:milk` and alike rather than the `forge` namespace for the fluids.
* A tag `forge:milk` was added containing the still and flowing milk like the vanilla water and lava tags. I am not sure if we want to add the tag to the `minecraft` namespace instead for consistency with the fluid's namespace, but I know some mods are using `forge:milk` for compat right now so it probably is better to stick with that.
* No fluid block is included to prevent the need for Forge to change the behavior of the vanilla bucket. I could add one that is simply unused by default if preferred.
* ~~I had to add a listener to the texture stitch event to ensure fluid textures are registered (KingLemming can give more details on why that is required). I am not sure if there is a better place than a client event listener subclass of `ForgeMod`, but I can say it does not cause issues when launching a server there.~~
* I did a bit of cleanup regarding milk in the bucket wrapper logic as the checks for milk are redundant to the fallback. The water and lava checks there really are redundant as well, but as they are just field lookups instead of object creation and are unrelated to this PR, I left them as is.

Included in this PR is `ForgeMilkTest` that will enable the Forge milk fluid and cause a milk bucket to display the name of the contained fluid on right click. You can enable the test mod using the boolean `ENABLE` flag.

Finally, credits to Rorax for the textures for the milk fluid. Used for this PR with permission.

### CODE REVIEW

**Comment by KnightMiner** [comment]
Updated to resolve conflicts with be58851

**Comment by KnightMiner** [comment]
Updated to resolve conflicts with c75eb9e

File: src/main/java/net/minecraftforge/common/ForgeMod.java [hunk 1]
```diff
@@ -229,4 +278,18 @@ public void registerRecipeSerialziers(RegistryEvent.Register<IRecipeSerializer<?
...
+        @SubscribeEvent
+        public static void registerFluidTextures(TextureStitchEvent.Pre event)
+        {
+            if (enableMilkFluid && PlayerContainer.LOCATION_BLOCKS_TEXTURE.equals(event.getMap().getTextureLocation()))
+            {
+                event.addSprite(STILL_MILK_TEXTURE);
+                event.addSprite(FLOWING_MILK_TEXTURE);
+            }
+        }
+    }
```
Comment (pupnewfster) [inline src/main/java/net/minecraftforge/common/ForgeMod.java]: Why is this needed? `FluidModel#getTextures` calls `ForgeHooksClient#getFluidMaterials` which gets the textures from the FluidAttributes.

Comment (KnightMiner) [reply src/main/java/net/minecraftforge/common/ForgeMod.java]: Is that new for 1.13+? (since the addition of vanilla fluids probably) I recall a discussion with KingLemming that this was required, and did not hear of it being resolved since then.

Comment (pupnewfster) [reply src/main/java/net/minecraftforge/common/ForgeMod.java]: The fluid system got rewritten in 1.14+ (1.13 didn't have one at all), and if memory serves giga made it so it auto adds the textures defined in FluidAttributes to the correct atlas so that modders don't have to have a bunch of boiler plate doing so.

Comment (KnightMiner) [reply src/main/java/net/minecraftforge/common/ForgeMod.java]: Removed the event listener

**Comment by Rosethorns** [comment]
I give my permission for Forge and all associated to use the milk fluid textures I created as linked.

**Comment by KnightMiner** [comment]
Updated for current Forge mappings on the main repo, and resolved conflicts with `ForgeMod` again.

**Comment by KingLemming** [comment]
This seems fine to me - were we waiting on anything else?

**Comment by LexManos** [comment]
No, It's fluids so you and cpw are the only one that touch it. If it doesn't break things I don't care.

**Comment by cpw** [comment]
Why is this PR a requirement?

**Comment by Buuz135** [comment]
Minecraft's milk bucket doesn't handle the fluid properly without a registry replacement and multiple modders might need milk as a fluid so this way we fix unnecesary fluid conflicts and registry replacements.

**Comment by cpw** [comment]
I approve and would have merged but the language changes need a local staging I cannot provide right now.
